contrib/NeonVM/neonvm.c

```diff
@@ -0,0 +1,820 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "opcode.h"
+#include "stack.h"
+
+#define error(s) fprintf(stderr, "%s\n", s); abort();
+
+typedef unsigned int uint32_t;
+typedef unsigned short uint16_t;
+
+static short get_uint16(const unsigned char *pobj, unsigned int nBuffSize, unsigned int *i)
+{
+    if (*i+2 > nBuffSize) {
+        error("Bytecode exception: Read past EOF.");
+    }
+    short r = (pobj[*i] << 8) | pobj[*i+1];
+    *i += 2;
+    return r;
+}
+
+static unsigned int get_uint32(const unsigned char *pobj, unsigned int nBuffSize, unsigned int *i)
+{
+    if (*i+4 > nBuffSize) {
+        error("Bytecode excpetion: Read past EOF.");
+    }
+    long r = (pobj[*i] << 24) | (pobj[*i+1] << 16) | (pobj[*i+2] << 8) | pobj[*i+3];
+    *i += 4;
+    return r;
+}
+
+struct NString {
+    unsigned int length;
+    char *data;
+};
+
+static unsigned char **getstrtable(const unsigned char *start, const unsigned char *end, unsigned int *cnt)
+{
+    unsigned char **r = NULL;
+    unsigned int i = 0;
+
+    *cnt = 0;
+    // We're going to iterate the string table first, to get the count,
+    // then we'll allocate the data.
+    const unsigned char *s = start;
+    while (s != end) {
+        s += ((s[0] << 24) | (s[1] << 16) | (s[2] << 8) | s[3]) + 4;
+        *cnt = *cnt + 1;
+    }
+
+    r = malloc((sizeof(unsigned char *)) * *cnt);
+    while (start != end) {
+        size_t strlen = (start[0] << 24) | (start[1] << 16) | (start[2] << 8) | start[3];

```

Comment:
Using `strlen` as a variable name (when it's already the name of a library function) is prone to confusion.